### PR TITLE
Add support for the DECST8C escape sequence

### LINKED ISSUE #16533: Add support for resetting tab stops (DECST8C)
# Description of the new feature/enhancement

[`DECST8C`](https://vt100.net/docs/vt510-rm/DECST8C.html) is an escape sequence which resets the tab stops to their default values, i.e. every 8 columns. It's especially useful for a modern terminal because it's the only practical way to restore the default values over an "infinite" width. Without this you'd have to set the positions manually, which limits you to the current screen width (technically you could also perform a hard reset with `RIS`, but that's not something an application should be doing).

# Proposed technical implementation details (optional)

We actually already have a method in `AdaptDispatch` that performs a tab reset (it's used in the hard reset implementation). We just need to make it public, and hook it up to the relevant escape sequence in the state machine.

It's a little bit more complicated than I'm making out, because the escape sequence takes a parameter value, and `DECST8C` is only applicable for one specific value, but that's easy enough to handle.

### PR DESCRIPTION
**Pull request by j4james**
## Summary of the Pull Request

This PR adds support for the `DECST8C` escape sequence, which resets the
tab stops to every 8 columns.

## Detailed Description of the Pull Request / Additional comments

This is actually a private parameter variant of the ANSI `CTC` sequence
(Cursor Tabulation Control), which accepts a selective parameter which
specifies the type of tab operation to be performed. But the DEC variant
only defines a single parameter value (5), which resets all tab stops.
It also considers an omitted parameter to be the equivalent of 5, so we
support that too.

## Validation Steps Performed

I've extended the existing tab stop tests in `ScreenBufferTests` with
some basic coverage of this sequence.

I've also manually verified that the `DECTABSR` script in #14984 now
passes the `DECST8C` portion of the test.

## PR Checklist
- [x] Closes #16533
- [x] Tests added/passed

### CODE REVIEW

File: src/terminal/adapter/adaptDispatch.cpp [hunk 1]
```diff
@@ -2793,16 +2793,22 @@ void AdaptDispatch::_ClearAllTabStops() noexcept
...
+// - setType - only SetEvery8Columns is supported
 // Return value:
-// - <none>
-void AdaptDispatch::_ResetTabStops() noexcept
+// - True if handled successfully. False otherwise.
+bool AdaptDispatch::TabSet(const VTParameter setType) noexcept
 {
-    _tabStopColumns.clear();
-    _initDefaultTabStops = true;
+    if (setType == DispatchTypes::TabSetType::SetEvery8Columns || !setType.has_value())
```
Comment (lhecker) [inline src/terminal/adapter/adaptDispatch.cpp:2805]: I feel like `value_or` would be perfect here in general, but trying to write such code it's not _that_ pretty:
```suggestion
    constexpr auto SetEvery8Columns = DispatchTypes::TabSetType::SetEvery8Columns;
    if (setType.value_or(SetEvery8Columns) == SetEvery8Columns)
```

I'll leave that up to you.

**Comment by lhecker** [comment]
BTW I didn't realize that `_tabStopColumns` uses a `std::vector<bool>`. Would it be alright to ask you to change it to a `std::vector<uint8_t>` or similar as part of this PR, to avoid the rather surprising bool-vector class?

**Comment by j4james** [comment]
> Would it be alright to ask you to change it to a `std::vector<uint8_t>` or similar as part of this PR, to avoid the rather surprising bool-vector class?

Done.

**Review by lhecker — APPROVED** [review]

**Review by zadjii-msft — APPROVED** [review]

**Review by DHowett — COMMENTED** [review]
Thanks for this!

**Pull request merged**